dusty/command_file.py

```diff
@@ -0,0 +1,129 @@
+    once_commands = app_spec['commands']["once"]
+    commands.append("then mkdir -p {}; touch {}".format(constants.RUN_DIR, first_run_file))
+    if once_commands:
+        commands += once_commands
+    commands.append("fi")
+    commands += app_spec['commands']['always']
+    return commands
+
+def _get_test_image_setup_commands(app_or_lib_name, expanded_specs, testing_spec):
+    commands = lib_install_commands_for_app_or_lib(app_or_lib_name, expanded_specs)
+    commands += ['cd {}'.format(container_code_path(spec_for_service(app_or_lib_name, expanded_specs)))]
+    commands += testing_spec['once']
+    return commands
+
+def _lib_install_commands_for_libs(assembled_specs, libs):
+    commands = []
+    for lib in libs:
+        lib_spec = assembled_specs['libs'][lib]
+        install_commands = _lib_install_commands(lib_spec)
+        if install_commands:
+            commands += install_commands
+    return commands
+
+def _lib_install_commands_for_app(app_name, assembled_specs):
+    """ This returns a list of all the commands that will install libraries for a
+    given app """
+    libs = assembled_specs['apps'][app_name]['depends']['libs']
+    return _lib_install_commands_for_libs(assembled_specs, libs)
+
+def _lib_install_commands_for_lib(app_name, assembled_specs):
+    """ This returns a list of all the commands that will install libraries for a
+    given lib """
+    libs = assembled_specs['libs'][app_name]['depends']['libs']
+    return _lib_install_commands_for_libs(assembled_specs, libs)
+
+def lib_install_commands_for_app_or_lib(app_or_lib_name, assembled_specs):
+    if app_or_lib_name in assembled_specs['apps']:
+        return _lib_install_commands_for_app(app_or_lib_name, assembled_specs)
+    return _lib_install_commands_for_lib(app_or_lib_name, assembled_specs)
+
+def _lib_install_commands(lib_spec):
+    """ This returns a single commmand that will install a library in a docker container """
+    if not lib_spec['install']:
+        return []
+    return ["cd {}".format(lib_spec['mount'])] + lib_spec['install']
+
+def dusty_command_file_name(app_or_lib_name, script_name=None, test_name=None):
+    command_file_name = 'dusty_command_file_{}'.format(app_or_lib_name)
+    if script_name:
+        command_file_name = '{}_script_{}'.format(command_file_name, script_name)
+    elif test_name:
+        command_file_name = '{}_test_{}'.format(command_file_name, test_name)
+    return "{}.sh".format(command_file_name)
+
+def _get_command_file_path(app_or_lib_name, assembled_specs, script_name=None, test_name=None):
+    if app_or_lib_name in assembled_specs['apps']:
+        spec = assembled_specs['apps'][app_or_lib_name]
+    else:
+        spec = assembled_specs['libs'][app_or_lib_name]
+    file_name = dusty_command_file_name(app_or_lib_name, script_name=script_name, test_name=test_name)
+    repo = Repo(spec['repo'])
+    return '{}/{}'.format(repo.local_path, file_name)
```

Comment:
We definitely shouldn't be writing files into the local copy of the repo. This would drive me insane. I get that we're trying to clean it up after, but that isn't guaranteed to happen and it's a hack anyway.

Ideally we could write the script and get it inside the container with a single exec call and some clever volume mounting.